client/src/core/TableClass.js

```diff
@@ -241,6 +251,112 @@ export class Table {
     return run_template(this.title_def[lang]);
   }
 
+  get dimensions() {
+    const cols = _.flatMap(this._cols, (col) =>
+      _.has(col, "children") ? col.children : col
+    );
+    const custom_dimensions = _.chain(cols)
+      .map("custom_groupings")
+      .compact()
+      .head()
+      .keys()
+      .value();
+    return _.concat(
+      "all",
+      _.chain(cols)
+        .filter((col) => col.groupings === undefined)
+        .map("nick")
+        .concat(custom_dimensions)
+        .compact()
+        .value()
+    );
+  }
+
+  is_custom_dim(dimension) {
+    return (
+      !_.chain(this._cols)
+        .flatMap((col) => (_.has(col, "children") ? col.children : col))
+        .map("nick")
+        .includes(dimension)
+        .value() && dimension != "all"
+    );
```

Comment:
Should be `!==` not `!=`.

Also, not massively important, but a minor optimization point. Putting the cheaper checks first is a free win, i.e. should short circuit and not perform the lodash chain here if `dimension !== "all"`.